app/views/contexts/summary_context.py

```diff
@@ -1,51 +1,133 @@
-                answer_store=self._answer_store,
-                list_store=self._list_store,
-                progress_store=self._progress_store,
-                metadata=self._metadata,
-                response_metadata=self._response_metadata,
-                current_location=location,
-                routing_path=self._router.routing_path(section_id),
-            )
-            section: Mapping = self._schema.get_section(section_id) or {}
-            if section.get("summary", {}).get("items"):
-                break
-
-            for group in section_summary_context(return_to=return_to)["summary"][
-                "groups"
-            ]:
-                yield group
+            if repeat := self._schema.get_repeat_for_section(section_id):
+                for_repeat = self._list_store[repeat["for_list"]]
+
+                if for_repeat.count > 0:
+                    for item in for_repeat.items:
+                        yield from self.build_summary_item(
+                            section_id=section_id,
+                            return_to=return_to,
+                            list_item_id=item,
+                            list_name=for_repeat.name,
+                        )
+            else:
+                yield from self.build_summary_item(
+                    section_id=section_id, return_to=return_to
+                )
+
+    def build_summary_item(
+        self,
+        section_id: str,
+        return_to: str | None,
+        list_name: str | None = None,
+        list_item_id: str | None = None,
+    ) -> list[dict]:
+        location = Location(
+            section_id=section_id, list_name=list_name, list_item_id=list_item_id
+        )
+        section_summary_context = SectionSummaryContext(
+            language=self._language,
+            schema=self._schema,
+            answer_store=self._answer_store,
+            list_store=self._list_store,
+            progress_store=self._progress_store,
+            metadata=self._metadata,
+            response_metadata=self._response_metadata,
+            current_location=location,
+            routing_path=self._router.routing_path(section_id),
+        )
+
+        groups: list = section_summary_context(
+            view_submitted_response=self.view_submitted_response, return_to=return_to
+        )["summary"].get("groups", [])
+
+        title_for_location = section_summary_context.title_for_location()
+        for group in groups:
+            if not list_item_id:
+                group["section_title"] = title_for_location
+            else:
+                item_title = (
+                    self._placeholder_renderer.render_placeholder(
+                        title_for_location, list_item_id
+                    )
+                    if isinstance(title_for_location, dict)
+                    else title_for_location
+                )
+                group["section_title"] = item_title
+
+        return groups
```

Comment:
This is odd as we are setting the section title, then unsetting it. You don't need to set the title on the group, you already have the title calling `section_summary_context()`. 

We could clean this up if we refactored to use sections of groups instead of a flattened list of group.

For now, instead of setting the title then unsetting it, just set it for the first one.


```py
        summary = section_summary_context(
            view_submitted_response=self.view_submitted_response, return_to=return_to
        )["summary"]
        groups: list = summary.get("groups", [])

        for group in groups:
            group["section_title"] = summary["title"]
            break

        return groups
```

Reply:
Updatted thos 👍